mlir/include/mlir/Dialect/MIOpen/LowerMIOpenOps.h

```diff
@@ -5169,16 +5191,21 @@ struct GridwiseGemmRewritePattern : public OpRewritePattern<miopen::GridwiseGemm
     int64_t GemmBBlockCopyThreadSliceLengths_GemmK;
     int64_t GemmBBlockCopyThreadSliceLengths_GemmN;
     assert(matrix_b_source_vector_read_dim != GemmG);
-    if (matrix_b_source_vector_read_dim == GemmK) {
+    switch (matrix_b_source_vector_read_dim) {
+    case GemmK:
       GemmBBlockCopyThreadSliceLengths_GemmK = matrix_b_source_data_per_read;
       GemmBBlockCopyThreadSliceLengths_GemmN =
           GemmBBlockCopyNumberDataPerThread /
           GemmBBlockCopyThreadSliceLengths_GemmK;
-    } else if (matrix_b_source_vector_read_dim == GemmMorN) {
+      break;
+    case GemmMorN:
       GemmBBlockCopyThreadSliceLengths_GemmN = matrix_b_source_data_per_read;
       GemmBBlockCopyThreadSliceLengths_GemmK =
           GemmBBlockCopyNumberDataPerThread /
           GemmBBlockCopyThreadSliceLengths_GemmN;
+      break;
+    case GemmG:
+      llvm_unreachable("Group matrix unimplemented");
```

Comment:
The message should be something like : " vector loads / stores are not possible on GemmG dimension ".